### PR TITLE
Flexible plots

### LINKED ISSUE #4140: start: update to use flexible plots
### Report

Need to update the get started pages to reflect https://github.com/iterative/example-repos-dev/pull/117

### PR DESCRIPTION
**Pull request by dberenbaum**
Closes #4140 

Depends on https://github.com/iterative/example-repos-dev/pull/117

### CODE REVIEW

**Comment by shcheklein** [comment]
@iterative/websites - should it be deployed?

**Comment by yathomasi** [comment]
Heroku has ended its free tier. It is asking for authorization from the admin.

Screenshot 2022-11-30 at 14 58 36

**Review by shcheklein — APPROVED** [review]

File: content/docs/start/data-management/metrics-parameters-plots.md [hunk 1]
```diff
@@ -34,25 +34,22 @@ attributes. Let's add a final evaluation stage to our
 [pipeline from before](/doc/start/data-management/data-pipelines):
```
Comment (jorgeorpinel) [inline content/docs/start/data-management/metrics-parameters-plots.md:34]: Let's add a final evaluation stage... -> add and run?

File: content/docs/start/data-management/metrics-parameters-plots.md [hunk 2]
```diff
@@ -62,33 +59,19 @@ evaluate:
...
-sections: `metrics` and `plots`. These are used to mark certain files containing
-ML "telemetry". Metrics files contain scalar values (e.g. `AUC`) and plots files
-contain matrices, data series (e.g. `ROC curves` or model loss plots), or images
-to be visualized and compared.
+The biggest difference to previous stages in our pipeline is the new `metrics`
+section. Metrics files contain scalar values (e.g. `AUC`) to compare across
+iterations.
 
 > With `cache: false`, DVC skips caching the output, as we want these JSON
 > metrics and plots files to be versioned by Git.
```
Comment (jorgeorpinel) [inline content/docs/start/data-management/metrics-parameters-plots.md:77]: Minor but good opportunity to use proper admons

```suggestion
<admon type="info">

With `cache: false`, DVC skips caching the output, as we want these JSON
metrics and plots files to be versioned by Git.

</admon>
```

File: content/docs/start/data-management/metrics-parameters-plots.md [hunk 3]
```diff
@@ -100,17 +83,40 @@ writes the model's
 [ROC-AUC](https://scikit-learn.org/stable/modules/model_evaluation.html#receiver-operating-characteristic-roc)
 and
 [average precision](https://scikit-learn.org/stable/modules/model_evaluation.html#precision-recall-and-f-measures)
-to `evaluation/metrics.json`, which in turn is marked as a `metrics` file with
-`-M`. Its contents are:
+for both train and test datasets to `eval/live/metrics.json`, which in turn is
+marked as a `metrics` file with `-M`. Its contents are:
```
Comment (jorgeorpinel) [inline content/docs/start/data-management/metrics-parameters-plots.md:87]: We could just do this...

```suggestion
marked as a [metrics file] with `-M`. Its contents are:

[metrics file]:/doc/command-reference/metrics#supported-file-formats
```

File: content/docs/start/data-management/metrics-parameters-plots.md [hunk 4]
```diff
@@ -100,17 +83,40 @@ writes the model's
...
+  "roc_auc": {
+    "train": 0.9873675866013153,
+    "test": 0.9619097316125981
+  }
+}
 ```
 
+> DVC doesn't force you to use any specific file names, nor does it enforce a
+> format or structure of a metrics file. It's completely user/case-defined.
+> Refer to `dvc metrics` for more details.
```
Comment (jorgeorpinel) [inline content/docs/start/data-management/metrics-parameters-plots.md:104]: ...and skip this whole note (probably TMI for GS).

```suggestion
```

File: content/docs/start/data-management/metrics-parameters-plots.md [hunk 5]
```diff
@@ -100,17 +83,40 @@ writes the model's
...
+
+You can view tracked metrics with DVC:
```
Comment (jorgeorpinel) [inline content/docs/start/data-management/metrics-parameters-plots.md:106]:
```suggestion
You can view tracked metrics with `dvc metrics show `:
```

File: content/docs/start/data-management/metrics-parameters-plots.md [hunk 6]
```diff
@@ -100,17 +83,40 @@ writes the model's
...
+$ dvc metrics show
+Path                    avg_prec.test    avg_prec.train    roc_auc.test    roc_auc.train
+eval/live/metrics.json  0.94496          0.97723           0.96191         0.98737
+```
+
+## Configuring plots
+
 `evaluate.py` also writes `precision`, `recall`, and `thresholds` arrays
 (obtained using
 [`precision_recall_curve`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_recall_curve.html))
```
Comment (jorgeorpinel) [inline content/docs/start/data-management/metrics-parameters-plots.md:118]: Unrelated but is this parenthesis necessary?

File: content/docs/start/data-management/metrics-parameters-plots.md [hunk 7]
```diff
@@ -100,17 +83,40 @@ writes the model's
...
-into the plots file `evaluation/plots/prc.json`:
+into the plots files `eval/prc/train.json` and `eval/prc/test.json`:
 
 ```json
```
Comment (jorgeorpinel) [inline content/docs/start/data-management/metrics-parameters-plots.md:121]: Q: Which one is seen in the sample below?

File: content/docs/start/data-management/metrics-parameters-plots.md [hunk 8]
```diff
@@ -124,38 +130,36 @@ into the plots file `evaluation/plots/prc.json`:
 
 Similarly, it writes arrays for the
 [roc_curve](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_curve.html)
-into `evaluation/plots/sklearn/roc.json`,
+and
 [confusion matrix](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html)
```
Comment (jorgeorpinel) [inline content/docs/start/data-management/metrics-parameters-plots.md:134]: 💅🏼 `roc_curve` and `confusion_matrix`

File: content/docs/start/data-management/metrics-parameters-plots.md [hunk 9]
```diff
@@ -124,38 +130,36 @@ into the plots file `evaluation/plots/prc.json`:
...
-`evaluation/plots/importance.png` with a feature importance bar chart for
-additional plots.
-
-> DVC doesn't force you to use any specific file names, nor does it enforce a
-> format or structure of a metrics or plots file. It's completely
-> user/case-defined. Refer to `dvc metrics` and `dvc plots` for more details.
-
-You can view tracked metrics and plots with DVC. Let's start with the metrics:
+into JSON files in the `eval/live/plots` directory, and an image
+`eval/importance.png` with a feature importance bar chart for additional plots.
```
Comment (jorgeorpinel) [inline content/docs/start/data-management/metrics-parameters-plots.md:136]: 💅🏼 and a `eval/importance.png` image showing...

Comment (jorgeorpinel) [reply content/docs/start/data-management/metrics-parameters-plots.md:136]: > for additional plots

What does that mean? I.e. are they saved to another plots file? Which one?

Comment (jorgeorpinel) [reply content/docs/start/data-management/metrics-parameters-plots.md:136]: p.s. More in general, if we could combine this paragraph with the previous one (before the code block), the section would be more readable: just a p and a code block before we get to the main point (configuring plots), next 👇🏼

File: content/docs/start/data-management/metrics-parameters-plots.md [hunk 10]
```diff
@@ -124,38 +130,36 @@ into the plots file `evaluation/plots/prc.json`:
...
 
-```cli
-$ dvc metrics show
-Path             avg_prec    roc_auc
-evaluation.json  0.89668     0.92729
-```
+To view plots, first configure the axes and other specifications of your plots
+by adding a `plots` section to your `dvc.yaml`:
```
Comment (jorgeorpinel) [inline content/docs/start/data-management/metrics-parameters-plots.md:139]: Could also simplify this and link to the spec, like:

```suggestion
Before generating plots, you must configure their [rendering properties]:

[rendering properties]: /doc/user-guide/project-structure/dvcyaml-files#plots
```

File: content/docs/start/data-management/metrics-parameters-plots.md [hunk 11]
```diff
@@ -124,38 +130,36 @@ into the plots file `evaluation/plots/prc.json`:
...
+      template: confusion
+      x: actual
+      y:
+        eval/live/plots/sklearn/cm/train.json: predicted
+        eval/live/plots/sklearn/cm/test.json: predicted
 ```
 
-Now let's view the plots. You can run `dvc plots show` on you terminal (shown
+Now let's view the plots. You can run `dvc plots show` on your terminal (shown
 below), which generates an HTML file you can open in a browser. Or you can load
```
Comment (jorgeorpinel) [inline content/docs/start/data-management/metrics-parameters-plots.md:163]: So should the section be called **Render plots** or **Configure and render plots**?

Also, 💅🏼 
```suggestion
To render them, you can run `dvc plots show` (shown below),
which generates an HTML file you can open in a browser. Or you can load
```

File: content/docs/start/data-management/metrics-parameters-plots.md [hunk 12]
```diff
@@ -124,38 +130,36 @@ into the plots file `evaluation/plots/prc.json`:
...
 your project in VS Code and use the [Plots Dashboard] of the [DVC Extension] to
 visualize them.
```
Comment (jorgeorpinel) [inline content/docs/start/data-management/metrics-parameters-plots.md:165]: Unrelated but I just noticed that this is redundant:

```suggestion
your project in VS Code and use the [Plots Dashboard] of the [DVC Extension].
```

**Review by jorgeorpinel — COMMENTED** [review]
Sorry, late review but the GS is pretty important so here are several comments. Some are minor (I'll start a follow-up PR) but there are a few good Qs too.

**Comment by jorgeorpinel** [comment]
Actually, only 2 Qs above remain (after https://github.com/iterative/dvc.org/pull/4207):

- https://github.com/iterative/dvc.org/pull/4145#discussion_r1056030858 and
- https://github.com/iterative/dvc.org/pull/4145#discussion_r1056031209 (well, I guess that's 2 Qs itself)

File: content/docs/start/data-management/metrics-parameters-plots.md (hunk 7 above)
Comment (dberenbaum) [reply content/docs/start/data-management/metrics-parameters-plots.md:121]: TBH I don't know. It's only an example of the file structure. Does it matter whether it shows train or test metrics?

File: content/docs/start/data-management/metrics-parameters-plots.md (hunk 9 above)
Comment (dberenbaum) [reply content/docs/start/data-management/metrics-parameters-plots.md:136]: Specifying all the plots files in one paragraph is a good idea. Do you want to add that to #4207?

File: content/docs/start/data-management/metrics-parameters-plots.md (hunk 7 above)
Comment (jorgeorpinel) [reply content/docs/start/data-management/metrics-parameters-plots.md:121]: No. I just made it clearer that it applies to both in the follow-up. Thanks

**Pull request merged**